### PR TITLE
[receiver/datadog]: follow receiver contract

### PR DESCRIPTION
**Pull request by VihasMakwana**
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35300 introduced a common methodology intended to be used by network receivers to follow [the contract](https://github.com/open-telemetry/opentelemetry-collector/blob/df3c9e38a80ccc3b14705462be2e2e51c628a3b3/receiver/doc.go#L10) and maintain the correct order of operations. This PR re-uses that for datadog.

### CODE REVIEW

**Review by MovieStoreGuy — COMMENTED** [review]
Two things that I had noticed, but otherwise seems reasonable.

File: receiver/datadogreceiver/receiver_test.go [hunk 1]
```diff
@@ -116,6 +117,55 @@ func TestDatadogServer(t *testing.T) {
...
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			cfg := createDefaultConfig().(*Config)
+			cfg.Endpoint = "localhost:0" // Using a randomly assigned address
+			dd, err := newDataDogReceiver(
+				cfg,
+				receivertest.NewNopSettings(),
+			)
+			dd.(*datadogReceiver).nextTracesConsumer = consumertest.NewErr(tc.err)
```
[the remark this answers is not in the document]
Comment (VihasMakwana) [reply receiver/datadogreceiver/receiver_test.go]: Done.

File: receiver/datadogreceiver/receiver_test.go [hunk 2]
```diff
@@ -116,6 +117,55 @@ func TestDatadogServer(t *testing.T) {
...
+
+			require.NoError(t, dd.Start(ctx, componenttest.NewNopHost()))
+			t.Cleanup(func() {
+				require.NoError(t, dd.Shutdown(ctx), "Must not error shutting down")
+			})
+
+			apiPayload := pb.TracerPayload{}
+			var reqBytes []byte
+			bytez, _ := apiPayload.MarshalMsg(reqBytes)
+			req, err := http.NewRequest(http.MethodPost, fmt.Sprintf("http://%s/v0.7/traces", dd.(*datadogReceiver).address), io.NopCloser(bytes.NewReader(bytez)))
```
[the remark this answers is not in the document]
Comment (VihasMakwana) [reply receiver/datadogreceiver/receiver_test.go]: Thanks for this! I've removed the wasted call and used [http.NewRequestWithContext](https://pkg.go.dev/net/http#NewRequestWithContext)

**Comment by VihasMakwana** [comment]
@MovieStoreGuy this is ready for another look!

**Review by MovieStoreGuy — APPROVED** [review]

**Pull request merged**